src/argilla/server/contexts/datasets.py

```diff
@@ -619,7 +619,7 @@ async def create_records(
     async with db.begin_nested():
         db.add_all(records)
         await db.flush(records)
-        await _load_users_from_record_responses(records)
+        await refresh_records(records)
         await search_engine.index_records(dataset, records)
```

Comment:
Not sure if it's the best approach but to explicitly link the preload of the records with the search engine maybe we can do the following:

```suggestion
        await search_engine.index_records(
            dataset, 
            await _preload_records_associations(records),
        )
```

Or if you don't want to do that:

```suggestion
        await _preload_records_associations(records)
        await search_engine.index_records(dataset, records)
```